### PR TITLE
better the the system test for global fit of water

### LINKED ISSUE #19249: better the system test for global fit of water
Existing system test `TeixeiraWaterSQESystemTest.py` did not catch the bug introduced by #19115 because there was no testing of the output fitted workspaces.

### PR DESCRIPTION
**Pull request by jmborr**
Added a check for output fitted workspace.

**To test:**
Just [run the test locally](http://www.mantidproject.org/System_Tests#Running_Tests_Locally)
```
cd your_build_directory/
./systemtest -R QENSGlobalFitTeixeiraWaterSQE
```

Fixes #19249.

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Testing/SystemTests/tests/analysis/QENSGlobalFitTeixeiraWaterSQE.py [hunk 1]
```diff
@@ -74,9 +75,20 @@ def runTest(self):
...
+            # check curves are correctly generated by calculating their Chi^2
+            residuals = np.empty(0)
+            for i in range(curves.size()):
+                curveset = curves[i]  # contains data, model, and residuals
+                dataY = curveset.dataY(0)
+                dataE = curveset.dataE(0)
+                modelY = curveset.dataY(1)
+                residuals = np.append(residuals, ((dataY-modelY)/dataE)**2)  # don't trust residuals of curveset
+            otherChi2 = residuals.sum()/len(residuals)
+            print otherChi2, chi2, abs(otherChi2-chi2)/chi2
```
Comment (rosswhitfield) [inline Testing/SystemTests/tests/analysis/QENSGlobalFitTeixeiraWaterSQE.py]: Can you fix the print statement for python 3, this systemtest previously passed with python 3 and we don't want to go backwards.

And add `from __future__ import (absolute_import, division, print_function)` to the top.

**Review by rosswhitfield — CHANGES_REQUESTED** [review]

File: Testing/SystemTests/tests/analysis/QENSGlobalFitTeixeiraWaterSQE.py (hunk 1 above)
Comment (jmborr) [inline Testing/SystemTests/tests/analysis/QENSGlobalFitTeixeiraWaterSQE.py]: Ouch, this I did for debugging purposes and forgot to remove it.

**Review by rosswhitfield — APPROVED** [review]

**Comment by rosswhitfield** [comment]
LGTM :shipit:

File: Testing/SystemTests/tests/analysis/QENSGlobalFitTeixeiraWaterSQE.py (hunk 1 above)
Comment (quantumsteve) [inline Testing/SystemTests/tests/analysis/QENSGlobalFitTeixeiraWaterSQE.py]: @jmborr Did you forget to add `from __future__ import (absolute_import, division, print_function)` to the top of the file?

Comment (jmborr) [inline Testing/SystemTests/tests/analysis/QENSGlobalFitTeixeiraWaterSQE.py]: Just fixed. Thanks for pointing it out

**Comment by jmborr** [comment]
Thank y'all

**Pull request merged**